clang/lib/Sema/Metafunctions.cpp

```diff
@@ -2007,7 +2007,15 @@ bool type_of(APValue &Result, Sema &S, EvalFn Evaluator, DiagFn Diagnoser,
 
   switch (RV.getReflectionKind()) {
   case ReflectionKind::Null:
-  case ReflectionKind::Type:
```

Comment:
I probably should have pointed to this earlier, but this is the spec for `type_of`:
https://brevzin.github.io/cpp_proposals/2996_reflection/d2996r6.html#pnum_210

`type_of(^^int)` fails to be a constant expression, so the existing behavior (i.e., diagnosing an error) is correct.

The cases that need modification are `Object`, `Value`, `Declaration`, `BaseSpecifier`, and `DataMemberSpec`.

(also for `return_type_of` below)

Ping me tomorrow if there's anything I can help with!